tests/Http/PostTest.php

```diff
@@ -1102,6 +1134,10 @@ public function testCreatingVoterRegistrationPost()
         // Mock the Blink API call.
         $this->mock(Blink::class)->shouldReceive('userSignupPost');
 
+        // Mock the Northstar API calls.
+            $this->mock(Registrar::class)
```

Comment:
Rather than mocking the `Registrar`, which has some Rogue-specific logic in it, let's mock the [`Northstar`](https://github.com/DoSomething/gateway/blob/15d3201b904d34d4220193938c0d4dce067f32dc/src/Northstar.php) client itself. We can set up a global mock in `createApplication` [like we do in Northstar](https://github.com/DoSomething/northstar/blob/e0eac7d94ec1b2c74345de61ac9a7ead1990ab11/tests/CreatesApplication.php#L46-L48):

```php
/**
 * Creates the application.
 *
 * @return \Illuminate\Foundation\Application
 */
public function createApplication()
{
    // ...

    $this->northstarMock = $this->mock(\DoSomething\Gateway\Northstar::class);

    $this->northstarMock->shouldReceive('asClient')->andReturnSelf();
    $this-> northstarMock->shouldReceive('getUser')->andReturnUsing(function ($type, $id) {
        return new NorthstarUser([
            // ...
        ]);
    });

```

Reply:
This also reminds me of [this pull request](https://github.com/DoSomething/gateway/pull/106), which I was pretty excited about back in the day.

Reply:
Ah this is great, thanks @DFurnes ! What would the `$type` param be here? Also, how would `createApplication()` receive the `$id` (or `$northstarId`)?

Reply:
The `$type` parameter is what kind of ID is being passed, so in this case [we pass `id`](https://github.com/DoSomething/rogue/blob/e1c4e3cb89df7b4d57809d422c5517768aa61c82/app/Services/Registrar.php#L33) and an ID, but we could theoretically also pass `'email'` or `'mobile'` (I don't think we do in Rogue, though, so not really necessary to cover in our mock.)

Reply:
> Also, how would createApplication() receive the $id (or $northstarId)?

It doesn't! The `createApplication` method is run by PHPUnit before each test case, and it configures a "fresh" instance of the Laravel app and sets up mocks or other things like that. 

The mock we've configured here for `Northstar` will then use this callback instead whenever some code calls `getUser` on it. (We're basically saying "Hey Laravel, whenever you're asked for `Northstar`, use this instead. And when this thing is asked to `getUser`, here's what it should do.")

Reply:
ahh I see, thank you for this explanation!!